honeybee_radiance_command/dctimestep.py

```diff
@@ -0,0 +1,277 @@
+    files, forming a matrix expression. The first argument is the View matrix file
+    that specifies how window output directions are related to some set of measured
+    values, such as an array of illuminance points or images.
+
+    Args:
+        options: Dctimestep options. It will be set to Radiance default values if
+            unspecified.
+        output: Path to the output matrix file. This should not be set if the output
+            file(s) has already been defined through the -o option in DctimestepOptions.
+        sky_vector: Path to the input sky_vector file, generated by gendaymtx or genskyvec
+            that is required for daylight coefficient, direct-daylight-coefficient
+            three-phase, three-phase-direct and four-phase-direct simulations.
+        sun_vector: Path to the input sun_vector file, generated by gendaymtx or genskyvec
+            that is required for daylight coefficient, direct-daylight-coefficient
+            three-phase, three-phase-direct and four-phase-direct simulations.
+        day_coef_matrix: Path to the input daylight coefficient matrix  file, generated
+            by rfluxmtx.
+        sun_coef_matrix: Path to the input sun_coefficient matrix file, generated by
+            rcontrib or rfluxmtx.
+        view_matrix: Path to the input view matrix, created by rfluxmtx, for three-phase,
+            four-phase, three-phase-direct and  four-phase-direct simulations.
+        t_matrix: Path to the input t (transmission matrix) BSDF file in xml format.
+        daylight_matrix: Path to the input daylight matrix, created by rfluxmtx, for
+            three-phase, four-phase, three-phase-direct and  four-phase-direct +
+            simulations.
+        facade_matrix: Path to the input facade matrix, created by rfluxmtx, for
+            four-phase and four-phase-direct simulations.
+
+    Properties:
+        * options
+        * output
+        * sky_vector
+        * sun_vector
+        * day_coef_matrix
+        * sun_coef_matrix
+        * facade_matrix
+        * view_matrix
+        * daylight_matrix
+        * t_matrix
+
+    """
+
+    __slots__ = ('_sky_vector', '_day_coef_matrix', '_view_matrix', '_daylight_matrix',
+                 '_t_matrix', '_sun_vector', '_sun_coef_matrix', '_facade_matrix')
+
+    def __init__(self, options=None, output=None, sky_vector=None, sun_vector=None,
+                 day_coef_matrix=None, sun_coef_matrix=None, view_matrix=None,
+                 t_matrix=None, daylight_matrix=None, facade_matrix=None):
+        """Initialize command"""
+        Command.__init__(self, output=output)
+        self.options = options
+        self.sky_vector = sky_vector
+        self.sun_vector = sun_vector
+        self.day_coef_matrix = day_coef_matrix
+        self.sun_coef_matrix = sun_coef_matrix
+        self.view_matrix = view_matrix
+        self.t_matrix = t_matrix
+        self.daylight_matrix = daylight_matrix
+        self.facade_matrix = facade_matrix
```

Comment:
This looks like a good case for providing several `classmethod` functions. It should also simplify the checks in your code.

Here is a pseudo-code. We can discuss this more tomorrow.

```suggestion
    def __init__(self, options=None, output=None, matrix_files=None):
        """Initialize command"""
        Command.__init__(self, output=output)
        self.options = options
        self.matrix_files = matrix_files or []
        
    @classmethod
    def from_daylight_coef(cls, options=None, output=None, day_coef_matrix=None, sky_vector=None):
        cls_ = cls(options, output, [daylight_coef_matrix, sky_vector]
        cls_._daylight_coeff = True  # I don't think we need this but just in case you think you should know which type of study you are dealing with
        return cls_
    
    
    @classmethod
    def from_sun_coef_matrix(cls, ...):
          ...
    
    @classmethod
    def from_view_matrix(cls, ...):
        ...

```

Reply:
The classmethods make a whole lot more sense than anything that I have been trying... I will implement the class this way and send the PR again.